semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -3056,6 +3057,85 @@ and m_directive a b =
   | G.OtherDirective _ ->
       fail ()
 
+and m_directive_vs_def a b =
+  let f (filea, importsa) = function
+    (* const x = require('y');
```

Comment:
I personally would just leave both what `a` and `b` should look like. So, for this case, I would say that this matches `const x = require('y')` to `import a`. It just takes me a little longer to notice that we're checking the value of `importsa`, and then I have to scroll to see what it means for `importsa` to be empty

Reply:
Good point